### ISSUE TITLE
XLSX Reader returns DateTime object from numeric value

### ISSUE THREAD

**Issue by konradkozaczenko** [box/spout#771]
Hi, in `Box\Spout\Reader\XLSX\Helper\CellValueFormatter.php` method `formatNumericCellValue` returns DateTime object even when $this->shouldFormatDates is set to false. In my xlsx file I have a cell with value "145280" and it is treated as DateTime which is wrong :/
StylesAttributes for that cell is 
numFmtId: 166
applyNumberFormat: null

Everything would be ok if here https://github.com/box/spout/blob/master/src/Spout/Reader/XLSX/Manager/StyleManager.php#L229 variable $applyNumberFormat would be compared to false with double equal sign instead of triple
... or check if shouldFormatDates is set to false earlier

**Comment by adrilo** [comment]
`$this->shouldFormatDates(false)` does return DateTime objects. A formatted date would be the string version, like "2020-05-02".
Adding a check `$applyNumberFormat === null` should do the trick I believe.

**Comment by adrilo** [comment]
Actually, this seems to be the right behavior. If `numFmtId` is set, `applyNumberFormat` does not need to be set to mean "do apply the number format, really".
So in your case, if `numFmtId 166` represents a date, then the correct behavior is to have Spout treat the number as a timestamp and return a DateTime object.

There's already a test covering this use case: https://github.com/box/spout/blob/2ff515c306b9c66d0ce3d84ea047a07562621d9f/tests/Spout/Reader/XLSX/ReaderTest.php#L353-L366 and I just verified that softwares apply the behavior I just described.
